tests/core/test_tracker_stores.py

```diff
@@ -876,3 +877,63 @@ def test_tracker_store_connection_error(config: Dict, domain: Domain):
 
     with pytest.raises(ConnectionException):
         TrackerStore.create(store, domain)
+
+
+def prepare_token_serialisation(
+    tracker_store: TrackerStore,
+    response_selector_interpreter: Interpreter,
+    sender_id: Text,
+):
+    text = "Good morning"
+    tokens = text.split()
+    current_index = 0
+    indices = []
+    for t in tokens:
+        start = current_index
+        end = current_index + len(t)
+        indices.append((start, end))
+        current_index = end + 1
+
+    tracker = tracker_store.get_or_create_tracker(sender_id=sender_id)
+    parse_data = response_selector_interpreter.parse(text)
+    event = UserUttered(
+        "Good morning",
+        parse_data.get("intent"),
+        parse_data.get("entities", []),
+        parse_data,
+    )
+
+    tracker.update(event)
+    tracker_store.save(tracker)
+
+    retrieved_tracker = tracker_store.retrieve(sender_id=sender_id)
+    event = retrieved_tracker.get_last_event_for(event_type=UserUttered)
+    event_tokens = event.as_dict().get("parse_data").get("text_tokens")
+
+    assert event_tokens[0][0] == indices[0][0]
+    assert event_tokens[0][1] == indices[0][1]
+    assert event_tokens[1][0] == indices[1][0]
+    assert event_tokens[1][1] == indices[1][1]
```

Comment:
I think you can check this in one line - `assert event_tokens == indices` - if you change `indices.append((start, end))` to `indices.append([start, end])`